### PR TITLE
Verify LIS modules load and version after LIS installation

### PR DESCRIPTION
**Pull request by sireeshapotnuru**
This PR is to add a check whether all the LIS modules are loaded or not after LIS installation along with build version check

### CODE REVIEW

File: Testscripts/Linux/LIS-MODULES-CHECK.py [hunk 1]
```diff
@@ -32,6 +33,23 @@ def RunTest(command):
...
+    [current_distro, distro_version] = DetectDistro()
+    lis_exists=Run("rpm -qa | grep hyper-v 2>/dev/null")
+
+    if distro_version.startswith("7") and lis_exists:
+        hvModules.append("pci_hyperv")
+        output = Run(command)
+        if not ("pci_hyperv" in output):
+            Run("modprobe pci_hyperv")
+
+    if (distro_version.startswith("7.3") or distro_version.startswith("7.4")) and lis_exists:
```
Comment (chvalean) [inline Testscripts/Linux/LIS-MODULES-CHECK.py:45]: you should not hardcode versions. at best put a greater than condition, if you know messages are to appear only from a version onwards.

File: Testscripts/Linux/VERIFY-LIS-MODULES-VERSION.sh [hunk 2]
```diff
@@ -53,6 +59,26 @@ for module in "${HYPERV_MODULES[@]}"; do
...
+if rpm -qa | grep hyper-v 2>/dev/null; then
+    expected_lis_version=$(dmesg | grep 'Vmbus LIS version' | awk -F ':' '{print $3}' | tr -d [:blank:])
+    if [[ $DISTRO_VERSION =~ 7\. ]]; then
+        HYPERV_MODULES+=('pci_hyperv')
+        pci_module=$(lsmod | grep pci_hyperv)
+        if [ -z $pci_module ]; then
+            modprobe pci_hyperv
+        fi
+    fi
+    if [[ $DISTRO_VERSION =~ 7\.3 ]] || [[ $DISTRO_VERSION =~ 7\.4 ]] ; then
```
Comment (chvalean) [inline Testscripts/Linux/VERIFY-LIS-MODULES-VERSION.sh:71]: ditto

File: Testscripts/Windows/LIS-INSTALL-SCENARIOS.ps1 [hunk 3]
```diff
@@ -8,6 +8,30 @@ param(
 )
 $ErrorActionPreference = "Stop"
 
+Function Check-modules() {
+    Write-LogInfo "Check if module are loaded after LIS installation"
+    $remoteScript = "LIS-MODULES-CHECK.py"
+    # Run the remote script
+    Run-LinuxCmd -username $user -password $password -ip $allVMData.PublicIP -port $allVMData.SSHPort -command "python ${remoteScript}" -runAsSudo
+    Run-LinuxCmd -username $user -password $password -ip $allVMData.PublicIP -port $allVMData.SSHPort -command "mv Runtime.log ${remoteScript}.log" -runAsSudo
+    Copy-RemoteFiles -download -downloadFrom $allVMData.PublicIP -files "/home/$user/state.txt,/home/${user}/${remoteScript}.log" -downloadTo $LogDir -port $allVMData.SSHPort -username $user -password $password
```
Comment (chvalean) [inline Testscripts/Windows/LIS-INSTALL-SCENARIOS.ps1:17]: line too long, split it with `

File: XML/Other/ReplaceableTestParameters.xml [hunk 4]
```diff
@@ -433,4 +433,8 @@ Scenario 2 : You need to run all the performance tests quickly.
 		<ReplaceThis>OVS_SOURCE_URL</ReplaceThis>
 		<ReplaceWith>https://github.com/openvswitch/ovs.git</ReplaceWith>
 	</Parameter>
+    <Parameter>
+		<ReplaceThis>EXPECTED_MLNX_VERSION</ReplaceThis>
+		<ReplaceWith>4.5-0.0.8</ReplaceWith>
```
Comment (chvalean) [inline XML/Other/ReplaceableTestParameters.xml:438]: this should be explained with a comment

**Review by chvalean — CHANGES_REQUESTED** [review]

**Comment by chvalean** [comment]
@iamshital @johnsongeorge-w are the current LIS deploy tests not checking already the LIS version?

**Comment by chvalean** [comment]
@sireeshapotnuru please post sample test results with both successful and failed cases, but maybe after the question/talk on this is done.

**Comment by ader1990** [comment]
@sireeshapotnuru make sure you reclone the repo (it has a new history) and cherry pick your commits to the new tree, push the changes and recreate the PR.

**Pull request closed without merging**